### ISSUE TITLE
BUG: something wrong at pd.Dataframe.from_dict()

### ISSUE THREAD

**Issue by zhenghao0379** [pandas-dev/pandas#31041]
#### Code Sample, a copy-pastable example if possible

```python
# Your code here
A = {"a":[1, 2, 3 ,4], "b":"b", "c":"c"}
pd.DataFrame.from_dict(A,orient='index').T

B = { "b":"b", "c":"c", "a":[1, 2, 3 ,4],}
pd.DataFrame.from_dict(B,orient='index').T

C = {"a":['1', '2', '3' ,'4'], "b":"b", "c":"c"}
pd.DataFrame.from_dict(C,orient='index').T
```
#### Problem description
the output A ，B  are diffrent 
and output C out error

**Comment by MarcoGorelli** [comment]
Thanks for the report.

This is because the data and index of the resulting DataFrame are determined by looking at the type of the first element in the dictionary:
```
                if isinstance(list(data.values())[0], (Series, dict)):
                    data = _from_nested_dict(data)
                else:
                    data, index = list(data.values()), list(data.keys())
```
It's assumed that each value in the dictionary should be of the same type. Core team - should this be mentioned in the docstring?

@zhenghao0379 if you change your example to 
```
A = {"a":[1, 2, 3 ,4], "b":["b"], "c":["c"]}
pd.DataFrame.from_dict(A,orient='index').T

B = { "b":["b"], "c":["c"], "a":[1, 2, 3 ,4],}
pd.DataFrame.from_dict(B,orient='index').T
```
then (barring column order) the output will be the same

**Comment by douglas125** [comment]
Shouldn't it be the case to create from nested if any of the elements is a list?

**Comment by MarcoGorelli** [comment]
@douglas125 I think that would be a possibility, but I presume that the core team decided against as it would mean having to check through the entire dictionary (so, evaluating the condition would require a linear - rather than constant - number of operations).

**Comment by gfyoung** [comment]
> I presume that the core team decided against as it would mean having to check through the entire dictionary instead of just the first value and that it seems reasonable to require people to pass values which are all of the same type.

Indeed, there would be performance implications, especially in a case where you have mixed inputs.  Not sure if that necessarily is deal breaker (it becomes linear in the number of inputs vs. constant time).  @jreback: Thoughts?

**Comment by MarcoGorelli** [comment]
@jreback sorry to ping you here - what action should we take? Should mixed inputs be supported?
